addons/goggles/functions/fnc_isDivingGoggles.sqf

```diff
@@ -1,22 +1,24 @@
 /*
- * Author: Garth 'L-H' de Wet
+ * Author: commy2
  * Determines whether passed goggles is diving goggles or a variant of them.
  *
  * Arguments:
  * 0: Glasses classname <STRING>
  *
  * Return Value:
- * Whether diving goggles are worn <BOOL>
+ * Check if these goggles are diving goggles <BOOL>
  *
  * Example:
- * [(goggles ace_player)] call ace_goggles_fnc_isDivingGoggles;
+ * [goggles ace_player] call ace_goggles_fnc_isDivingGoggles;
  *
  * Public: Yes
  */
 #include "script_component.hpp"
-private ["_result", "_glasses"];
-_glasses = _this select 0;
-_result = _glasses == "G_Diving";
-if (_result) exitWith {true};
-_result = [configFile >> "CfgGlasses" >> _glasses, configFile >> "CfgGlasses" >> "G_Diving"] call CBA_fnc_inheritsFrom;
-_result
+
+params ["_glasses"];
+
+local _config = configFile >> "CfgGlasses" >> _glasses;
+
+if (!isClass _config) exitWith {false};
+
+getNumber (_config >> "mode") == 1 // return
```

Comment:
`mode = 1;` is a config entry that changes the behaviour of goggles.
Usually goggles are hidden on the unit if it gets into the water even though they are equipped in that slot.
That is reversed when `mode` is set to 1. The goggles are only shown when in water. This is used for diving goggles in vanilla.